backend/ttrs/views.py

```diff
@@ -150,3 +150,19 @@ class MajorDetail(generics.RetrieveAPIView):
     queryset = Major.objects.all()
     serializer_class = MajorSerializer
     permission_classes = (AllowAny,)
+
+
+class RecommendView(generics.ListCreateAPIView):
```

Comment:
I think this view does not need to inherit `ListMixin`.
It would be more simple to inherit just `RetrieveAPIView` like this,
```python3
class RecommandView(generics.RetrieveAPIView):
```
`RetrieveAPIView` gets GET method requests.
The reason I make the view get only GET request is, I think there is no sensible data sent from the student.
So, you can send data(options) in the url by GET method.

For more specific usage, please refer to the code of `FilterAPIView` which I have implemented before.